libc/harness/test_exec.py

```diff
@@ -0,0 +1,109 @@
+# Phoenix-RTOS
+#
+# libc-tests
+#
+# various exec functions tests
+#
+# Copyright 2021 Phoenix Systems
+# Author: Damian Loewnau
+#
+# This file is part of Phoenix-RTOS.
+#
+# %LICENSE%
+#
+
+from psh.tools.basic import run_psh, assert_only_prompt, assert_expected
+
+
+def assert_execve_env_changed(p):
+    cmd = '/bin/test_exec 1'
+    expected = (r'(argc = 1(\r+)\n)'
+                r'(argv\[0\] = /bin/to_exec(\r+)\n)'
+                r'(environ\[0\] = TEST1=exec_value(\r+)\n)'
+                r'(\r\x1b\[0J\(psh\)%)')
+
+    assert_expected(p, cmd, expected, "Wrong output of execve function with changed environment")
+
+
+def assert_execve_env_unchanged(p):
+    cmd = '/bin/test_exec 2'
+    expected = (r'(argc = 1(\r+)\n)'
+                r'(argv\[0\] = /bin/to_exec(\r+)\n)'
+                r'(environ\[0\] = TEST1=unchanged_value(\r+)\n)'
```

Comment:
this assumes we will have empty ENV when starting this test (which might not be the case), Maybe it would be better just to check if `TEST1=unchanged_value` exists somewhere in the environment? (so - more complicated regexp).

The alternative approach is to use `clearenv()` explicitly in C code.

Reply:
Thanks for review, changes applied. I will add more test cases in the future.

Reply:
ok, please copy my ideas for new tests to JIRA issue not to forget them in the future :).